docs/operations/clean-metadata-store.md

```diff
@@ -117,12 +115,19 @@ Rule cleanup uses the following configuration:
 
 ### Compaction configuration records
 
+Druid retains all compaction configuration records by default, which should be suitable for most use cases.
+If you have a high datasource churn rate, that is, you create and delete short-lived datasources with high frequency, consider turning on automated cleanup of compaction configuration records.
+With this feature turned on, when you create a compaction configuration for some datasource before the datasource exists, for example if ingestion is ongoing, Druid may remove the compaction configuration.
+To prevent the configuration from being prematurely removed, wait to set `druid.coordinator.kill.compaction.on=true` until after the datasource is created.
```

Comment:
```suggestion
To prevent the configuration from being prematurely removed, wait to set auto compaction configuration on the datasource after the datasource is created.
```